json-rpc/tests/integration_test.rs

```diff
@@ -598,6 +601,45 @@ fn create_test_cases() -> Vec<Test> {
                 );
             },
         },
+        Test {
+            name: "upgrade event",
+            run: |env: &mut testing::Env| {
+                let txn = env.create_txn_by_payload(
+                    &env.root,
+                    TransactionPayload::WriteSet(WriteSetPayload::Direct(ChangeSet::new(
+                        create_common_write_set(),
+                        vec![],
+                    ))),
+                );
+                let result = env.submit_and_wait(txn);
+                let version = result["version"].as_u64().unwrap();
+                assert_eq!(
+                    result["events"],
+                    json!([
+                        {
+                            "data":{
+                                "type": "upgrade",
+                                "write_set": "01c4c63f80c74b11263e421ebf8486a4e30901217da6c6b3e19f180104cafed00d00"
```

Comment:
Do we want to add an extra test to make sure the writeset here is equal to the lcs serialized common writeset?